### PR TITLE
fix(admin-ui): inventory item deletion when removing product

### LINKED ISSUE #5719: Deleting products leaves orphan entries in inventory
### Describe the bug

When I delete a product on medusa 1.18 with the enabled inventory plugin, it leaves orphaned objects in the inventory tab which can't be deleted.

### System information

Medusa version (including plugins): v1.18.0
Node.js version: 20.10.0
Database: Postgres
Operating system: ubuntu server
Browser (if relevant):

### Steps to reproduce the behavior

1. Add product, set "manage inventory" to true, add certain amount
2. Publish
3. Delete
4. Check inventory tab, a product with no image or name is shown with inventory

### Expected behavior
The inventory is also deleted/can be deleted

### Screenshots

Bildschirmfoto 2023-11-23 um 21 18 49

### PR DESCRIPTION
**Pull request by pKorsholm**
**What**
- remove inventory items when deleting the product

**How**
- Like we've handled all inventory interactions in admin so far

closes #5719

### CODE REVIEW

File: packages/admin-ui/ui/src/components/templates/product-table/use-product-actions.tsx [hunk 1]
```diff
@@ -32,6 +43,24 @@ const useProductActions = (product: Product) => {
     })
 
     if (shouldDelete) {
+      if (isFeatureEnabled("inventoryService")) {
+        const { variants } = await client.admin.variants.list({
+          id: product.variants.map((v) => v.id),
+          expand: "inventory_items",
+        })
+
+        console.warn("variants", variants)
```
Comment (olivermrbl) [inline packages/admin-ui/ui/src/components/templates/product-table/use-product-actions.tsx]: todo: rm

Comment (kasperkristensen) [inline packages/admin-ui/ui/src/components/templates/product-table/use-product-actions.tsx]: rm

File: packages/admin-ui/ui/src/components/templates/product-table/use-product-actions.tsx [hunk 2]
```diff
@@ -1,17 +1,25 @@
...
+import { getErrorMessage } from "../../../utils/error-messages"
 import useCopyProduct from "./use-copy-product"
+import { useFeatureFlag } from "../../../providers/feature-flag-provider"
+import useImperativeDialog from "../../../hooks/use-imperative-dialog"
+import { useNavigate } from "react-router-dom"
+import useNotification from "../../../hooks/use-notification"
+import { useQueryClient } from "@tanstack/react-query"
+import { useTranslation } from "react-i18next"
 
 const useProductActions = (product: Product) => {
```
Comment (olivermrbl) [inline packages/admin-ui/ui/src/components/templates/product-table/use-product-actions.tsx:24]: todo: We should also delete inventory items when you delete a product from the details page. This is only for the table action, as far as I can see

**Review by olivermrbl — CHANGES_REQUESTED** [review]

**Review by olivermrbl — APPROVED** [review]

**Pull request merged**